crates/solver/src/settlement_submission.rs

```diff
@@ -176,9 +176,24 @@ impl SolutionSubmitter {
             }
         }
 
+        // combine protocol enabled strategies with settlement prefered strategies
+        let strategies = match settlement.submitter {
+            SubmissionPreference::ProtocolDefault => self.transaction_strategies.as_slice(),
+            SubmissionPreference::Flashbots => {
+                let position = self
+                    .transaction_strategies
+                    .iter()
+                    .position(|strategy| matches!(strategy, TransactionStrategy::Flashbots(_)));
```

Comment:
I think this simple approach is a great start!

One issue I foresee is that this requires Flashbots to be enabled as a strategy - which affects all of our settlements and not just the ones that have `SubmissionPreference::Flashbots`. We currently have this disabled, but is this something we plan on enabling generally?

Reply:
For now, this is a more safe/strict solution, meaning solvers can express their preference only for strategies that are currently enabled by the protocol.
But we can definitely change this logic and give searchers more power - to override whatever protocol decided. However I was not sure we want this, because if we do and Flashbots are slow, it would affect our protocol throughput (at least until driver solver colocation is done where we have defined deadlines, penalties and so on)

Reply:
What I was trying to suggest is that I don't think we can enable Flashbots without also routing all settlements through Flashbots (i.e. I'm not sure how we configure the solver so only settlements with `SubmissionPreference::Flashbots` use Flashbots).

The only way I could think of solving it would be have both a `--supported-submission-strategies` and a `--default-submission-strategies` flag.

Reply:
(to be clear, I think this can be done in a **follow up PR** - just think this might be needed to achieve our desired configuration).